### PR TITLE
Add option to use permissions boundary on the default roles

### PR DESCRIPTION
**Pull request by 0xVox**
## what
* Allow the use of a permissions boundary on the default role

## why

* In our organisation, permissions boundaries are a requirement for all engineers to provide to roles. Allowing them to be attached to the default roles allows us to create node groups without creating our own roles and policies.

* The change is entirely optional and non-breaking.

## references
* The eks_cluster module you provide also allows this.

### CODE REVIEW

**Review by Gowiem — DISMISSED** [review]
One change, but other will be good to merge. Update and I'll review.

File: variables.tf [hunk 1]
```diff
@@ -413,6 +413,12 @@ variable "use_existing_aws_iam_instance_profile" {
   default     = false
 }
 
+variable "aws_iam_role_permissions_boundary" {
```
[the remark this answers is not in the document]
Comment (nitrocode) [reply variables.tf:416]: agreed

```suggestion
variable "permissions_boundary" {
```

File: main.tf [hunk 2]
```diff
@@ -34,10 +34,11 @@ data "aws_iam_policy_document" "assume_role" {
...
 resource "aws_iam_role" "default" {
-  count              = local.enabled && var.use_existing_aws_iam_instance_profile == false ? 1 : 0
-  name               = module.label.id
-  assume_role_policy = join("", data.aws_iam_policy_document.assume_role.*.json)
-  tags               = module.label.tags
+  count                = local.enabled && var.use_existing_aws_iam_instance_profile == false ? 1 : 0
+  name                 = module.label.id
+  assume_role_policy   = join("", data.aws_iam_policy_document.assume_role.*.json)
+  tags                 = module.label.tags
+  permissions_boundary = var.aws_iam_role_permissions_boundary
```
Comment (nitrocode) [inline main.tf:41]:
```suggestion
  permissions_boundary = var.permissions_boundary
```

**Review by Gowiem — APPROVED** [review]

**Comment by Gowiem** [comment]
Thanks for the assist @nitrocode 👍  -- I need to remember that I can do that for simple changes like I mentioned.

**Comment by 0xVox** [comment]
Thanks all :)

**Pull request merged**